### ISSUE TITLE
warn on uninitialized value passed by const reference

### ISSUE THREAD

**Issue by d0k** [llvm/llvm-project#12299]
|  |  |
| --- | --- |
| Bugzilla Link | [11927](https://llvm.org/bz11927) |
| Version | trunk |
| OS | All |
| CC | @DougGregor,@tkremenek |

## Extended Description 
int ref(const int&);

void test1() {
  int x;
  ref(x);
}

I think clang should warn about passing x uninitialized by const reference here. "ref" can modify the value by casting the const away, but that's not really something worth encouraging.

**Comment by wheatman** [comment]
This seems to have been fixed between 10.0.1 and 11.0.0
There is now a warning which shows up with `-Wall`
https://godbolt.org/z/oGn5cKcTE
```
<source>:5:5: warning: variable 'x' is uninitialized when passed as a const reference argument here
 [-Wuninitialized-const-reference]
ref(x);
```
